src/rt/test/func/runtimepause/runtimepause.cc

```diff
@@ -28,53 +29,72 @@ struct M : public VBehaviour<M>
   }
 };
 
-void test_runtime_pause(size_t cores, size_t pauses)
+struct Dummy : public VCown<Dummy>
+{};
+
+struct RemoveExternalEvent : public VBehaviour<RemoveExternalEvent>
 {
-  Scheduler& sched = Scheduler::get();
-  sched.init(cores);
-  Scheduler::add_external_event_source();
+  Cown* a;
 
-  auto a = new A;
+  RemoveExternalEvent(Cown* a) : a(a) {}
 
-  auto thr = std::thread([pauses, &a]() mutable {
-    std::mt19937 rng;
-    rng.seed(1);
-    std::uniform_int_distribution<> dist(1, 1000);
-    for (size_t i = 1; i <= pauses; i++)
-    {
-      auto pause_time = std::chrono::milliseconds(dist(rng));
-      std::this_thread::sleep_for(pause_time);
-      Cown::schedule<M>(a, a, i, i == pauses);
-    }
+  void f()
+  {
+    Systematic::cout() << "Remove external event source" << std::endl;
+    Scheduler::remove_external_event_source();
+    Cown::release(ThreadAlloc::get(), a);
+  }
+};
 
-    auto pause_time = std::chrono::nanoseconds(dist(rng));
-    std::this_thread::sleep_for(pause_time);
+struct StartExternalThread : public VBehaviour<StartExternalThread>
+{
+  size_t pauses;
 
-    Scheduler::remove_external_event_source();
-  });
+  StartExternalThread(size_t pauses) : pauses(pauses) {}
 
-  sched.run();
-  thr.join();
-  snmalloc::current_alloc_pool()->debug_check_empty();
+  void f()
+  {
+    auto a = new A;
+    auto pauses_ = pauses;
```

Comment:
Unless I'm mistaken, this new variable `pauses_` isn't necessary since a lambda capture of `pauses` will copy the value.

Reply:
`pauses` is a field of the messages, so can't appear in the capture list.  I could capture, `this` but it has the wrong lifetime.